test/esbuild.js

```diff
@@ -334,5 +334,33 @@ test('xdm (esbuild)', async function (t) {
 
   console.log('\nnote: the preceding errors and warnings are expected!\n')
 
+  /** @type import('esbuild').Plugin */
+  const inlinePlugin = {
+    name: 'inline plugin',
+    setup: (build) => {
+      build.onResolve({filter: /.*/}, () => {
+        return {
+          path: path.join(process.cwd(), 'index.mdx'),
+          pluginData: {
+            contents: `# Test
+            Test Contents
+            `.trim()
+          }
+        }
+      })
+    }
+  }
+
+  const buildResult = await esbuild.build({
+    entryPoints: [path.join(process.cwd(), 'index.mdx')],
+    plugins: [inlinePlugin, esbuildXdm()],
+    write: false
+  })
+
+  const decoder = new StringDecoder('utf8')
+  const code = decoder.write(Buffer.from(buildResult.outputFiles[0].contents))
+
+  t.match(code, /children: "Test"\n/)
```

Comment:
This ia pretty simple test. Not sure if there is a better way to confirm that xdm built it properly.

Reply:
A simple test is fine but this looks a bit frail indeed

How about something done in the rest of the file: letting esbuild write to a certain file, importing that file, asserting what in results in and then unlinking it? That makes sure all the syntax is valid, that it loads, and what it yields

https://github.com/Arcath/xdm/blob/0be96728f4baa1146662ee8e2b2249c9d94481bf/test/esbuild.js#L92-L103